sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -161,16 +161,49 @@ public HttpTransport(
    */
   @Override
   public boolean isRetryAfter(final @NotNull String type) {
-    if (sentryRetryAfterLimit.containsKey(type)) {
-      final Date date = sentryRetryAfterLimit.get(type);
+    final String category = getCategoryFromType(type);
+    final Date currentDate = new Date();
+
+    // check all categories
+    if (sentryRetryAfterLimit.containsKey(HTTP_RETRY_ALL_CATEGORIES)) {
+      final Date date = sentryRetryAfterLimit.get(HTTP_RETRY_ALL_CATEGORIES);
+      if (!currentDate.after(date)) {
+        return true;
+      }
+    }
+    // check for specific category
+    if (sentryRetryAfterLimit.containsKey(category)) {
+      final Date date = sentryRetryAfterLimit.get(category);
+      return !currentDate.after(date);
+    }
 
-      return !new Date().after(date);
-    } else if (sentryRetryAfterLimit.containsKey(HTTP_RETRY_DEFAULT_CATEGORY)) {
-      final Date date = sentryRetryAfterLimit.get(HTTP_RETRY_DEFAULT_CATEGORY);
+    return false;
+  }
+
+  /**
+   * Returns a rate limiting category from type
+   *
+   * @param type the type (eg event, session, attachment, ...)
+   * @return the category eg (error, session, attachment)
+   */
+  private @NotNull String getCategoryFromType(final @NotNull String type) {
```

Comment:
I think we have two different mappings. One for `events` and one for `envelopeItems`. Is this still valid @jan-auer? If yes, can we name the method and parameter either `envelopeType` or `eventType` to avoid misinterpretation?

Reply:
I guess you are mixing up, event types or envelope types is a way to define the type of something, eg a crash event is the type of `event` and category `error`.
if a crash event is sent by an envelope or not, its still a `event` as a type and `error` as a category.

also, please use our logaf scale